### PR TITLE
Fix E2E tests

### PR DESCRIPTION
**Pull request by thingalon**
This PR fixes E2E test instability.

#### Changes proposed in this Pull Request:
* Mock out speed scores, except for tests which explicitly rely on them - to reduce system stress
* Restart speed score requests which wpcom doesn't recognize
* Update the way that connected state is detected not to be so reliant on race-conditions.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Run the e2e test suite!

### CODE REVIEW

**Review by eAdnan007 — COMMENTED** [review]
I still have a few failed tests when I run this pr branch.

```
5 failed
    specs/common.test.js:17:1 › Click on the plugins page should navigate to Boost settings page ===
    specs/common.test.js:32:1 › Deactivating the plugin should clear Critical CSS and Dismissed Recommendation notice option 
    specs/connection.test.js:7:2 › Settings Page Connection › Should connect to WP.com on a fresh install with Jetpack plugin activated and Jetpack already connected 
    specs/connection.test.js:18:2 › Settings Page Connection › Should connect to WP.com on a fresh install with Jetpack plugin activated 
    specs/critical-css.test.js:65:2 › Critical CSS module › Critical CSS Admin message should show when the theme is changed 
```

File: projects/plugins/boost/app/lib/class-speed-score-request.php [hunk 1]
```diff
@@ -196,18 +196,20 @@ public function poll_update() {
 		);
 
 		if ( is_wp_error( $response ) ) {
+			// Special case: If the request is not found, restart it.
+			if ( 'not_found' === $response->get_error_code() ) {
+				return $this->restart();
```
Comment (pyronaur) [inline projects/plugins/boost/app/lib/class-speed-score-request.php:201]: Can this cause an infinite loop of restarts? Or is that the intention?

**Review by pyronaur — APPROVED** [review]
LGTM.

I do have one question in there though.

**Comment by pyronaur** [comment]
Oh shoot. I'm seeing the same broken tests as @eAdnan007 in my Critical CSS PR 😬  - I thought it was broken because of the PR but it seems like the tests really are still a bit broken.

File: projects/plugins/boost/app/lib/class-speed-score-request.php (hunk 1 above)
Comment (thingalon) [reply projects/plugins/boost/app/lib/class-speed-score-request.php:201]: 19c3d99727 adds a limit to how often it's allowed to retry.

**Review by karthikax — DISMISSED** [review]
A few tests fail for me as well (Different than what Adnan posted).
However, since this pull request mainly improves the test run by mocking speed score requests, I am happy to move forward with the changes!

**Comment by thingalon** [comment]
043c3bb -- yeah, that's probably a better approach.

**Comment by thingalon** [comment]
Tests are all regularly passing on:
- My macbook (both PHP 7.4 and PHP 8.1)
- An unbuntu vps
- Github.

**Comment by thingalon** [comment]
Adnan's latest changes look good to me.

**Review by pyronaur — APPROVED** [review]
👍 

Thanks for figuring these out @thingalon 

Looks good to ship! 🚢

**Review by adimoldovan — APPROVED** [review]
Looks good! 🚢

**Pull request merged**